lib/addict/controllers/base_controller.ex

```diff
@@ -0,0 +1,113 @@
+defmodule Addict.BaseController do
+  @moduledoc """
+    Addict BaseController is used as a base to be extended by controllers if needed.
+    BaseController has functions to receive User related requests directly from
+    the Phoenix router.  Adds `register/2`, `logout/2` and `login/2` as public functions.
+   """
+
+  defmacro __using__(_) do
+    quote do
+      @manager Application.get_env(:addict, :manager) || Addict.ManagerInteractor
+      alias Addict.SessionInteractor
+      alias Addict.Utils
+
+      @doc """
+      GET paths for register, login, password recovery, and password reset
+      """
+      def register(conn, _params), do: render_action(conn)
+      def login(conn, _params), do: render_action(conn)
+      def recover_password(conn, _params), do: render_action(conn)
+      def reset_password(conn, _params), do: render_action(conn)
+
+      defp render_action(conn) do
+        addict = conn.private.addict
+
+        layout = case addict.layout do
+          false ->
+            false
+          nil ->
+            {Module.concat([Utils.app_name(conn), LayoutView]), "application.html"}
+          _ ->
+            addict.layout
+        end
```

Comment:
It'd be great to extract this `case` into a method to make it even more explicit.